lms/djangoapps/courseware/user_state_client.py

```diff
@@ -53,12 +53,7 @@ def __init__(self, user=None):
         """
         self.user = user
 
-    @contract(
-        username="basestring",
-        block_key=UsageKey,
-        scope=ScopeBase,
-        fields="seq(basestring)|set(basestring)|None"
-    )
+    @donottrack(StudentModule, StudentModuleHistory)
```

Comment:
`@contract` created a argument specs conflict with `@donottrack`, and I am quite sure that it will conflict with any kind of parameterized decorator . 
Reason to remove `@contract` in this file -
We have declared it in `edx-user-state-client/edx-user-state-client/interface.py:XBlockUserStateClient`